dbclient/tracing/src/main/java/io/helidon/dbclient/tracing/DbClientTracing.java

```diff
@@ -77,42 +75,41 @@ protected Single<DbClientServiceContext> apply(DbClientServiceContext serviceCon
         }
 
         Context context = serviceContext.context();
-        Tracer tracer = context.get(Tracer.class).orElseGet(GlobalTracer::get);
+        Tracer tracer = context.get(Tracer.class).orElseGet(Tracer::global);
 
         // now if span context is missing, we build a span without a parent
-        Tracer.SpanBuilder spanBuilder = tracer.buildSpan(serviceContext.statementName());
+        Span.Builder<?> spanBuilder = tracer.spanBuilder(serviceContext.statementName());
 
         context.get(SpanContext.class)
-                .ifPresent(spanBuilder::asChildOf);
+                .ifPresent(spanBuilder::parent);
 
         Span span = spanBuilder.start();
 
-        span.setTag("db.operation", serviceContext.statementType().toString());
+        span.tag("db.operation", serviceContext.statementType().toString());
         if (spanConfig.logEnabled("statement", true)) {
-            Tags.DB_STATEMENT.set(span, serviceContext.statement());
+            Tag.DB_STATEMENT.create(serviceContext.statement()).apply(span);
         }
-        Tags.COMPONENT.set(span, "dbclient");
-        Tags.DB_TYPE.set(span, serviceContext.dbType());
+        Tag.COMPONENT.create("dbclient").apply(span);
+        Tag.DB_TYPE.create(serviceContext.dbType()).apply(span);
 
         serviceContext.statementFuture().thenAccept(nothing -> {
             if (spanConfig.logEnabled("statement-finish", true)) {
-                span.log(Map.of("type", "statement"));
+                span.addEvent("log", Map.of("type", "statement"));
             }
         });
 
         serviceContext.resultFuture().thenAccept(count -> {
             if (spanConfig.logEnabled("result-finish", true)) {
-                span.log(Map.of("type", "result",
+                span.addEvent("log", Map.of("type", "result",
                                 "count", count));
             }
-            span.finish();
+            span.end();
         }).exceptionally(throwable -> {
-            Tags.ERROR.set(span, Boolean.TRUE);
-            span.log(Map.of("event", "error",
-                            "error.kind", "Exception",
+            Tag.ERROR.create(Boolean.TRUE).apply(span);
```

Comment:
Please replace the whole thing with `span.end(throwable)` - this is all handled already (see `OpenTracingSpan.end(Throwable)`)

Reply:
done